src/Language/PureScript/Errors.hs

```diff
@@ -763,16 +764,24 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
       line $ "Type of expression contains additional label " <> markCode (prettyPrintLabel prop) <> "."
     renderSimpleErrorMessage TypeSynonymInstance =
       line "Type class instances for type synonyms are disallowed."
-    renderSimpleErrorMessage (OrphanInstance nm cnm ts) =
-      paras [ line $ "Type class instance " <> markCode (showIdent nm) <> " for "
+    renderSimpleErrorMessage (OrphanInstance nm cnm nonOrphanModules ts) =
+      paras [ line $ "Orphan instance " <> markCode (showIdent nm) <> " found for "
             , markCodeBox $ indent $ Box.hsep 1 Box.left
                 [ line (showQualified runProperName cnm)
                 , Box.vcat Box.left (map typeAtomAsBox ts)
                 ]
-            , line "is an orphan instance."
-            , line "An orphan instance is one which is defined in a module that is unrelated to either the class or the collection of data types that the instance is defined for."
-            , line "Consider moving the instance, if possible, or using a newtype wrapper."
-            ]
+            , Box.vcat Box.left $ case modulesToList of
+                [] -> [ line "There is nowhere this instance can be placed without being an orphan."
+                      , line "A newtype wrapper can be used to avoid this problem."
+                      ]
+                _  -> [ Box.text $ "This instance must be declared in "
+                          <> T.unpack formattedModules
+                          <> ", or be defined for a newtype wrapper."
```

Comment:
I'm wondering about the phrasing here; the use of 'must' makes it sound like these are strictly the only two possible solutions.  However, there are potentially many other contextually dependent solutions.

Reply:
https://github.com/purescript/purescript/pull/3106/commits/c724145c3aa8c557ea2f01a0883b635cf43e35f3#diff-2a67dc7a3cb0d6822d66486323cc9420R777 - what do you think?